src/vaadin-grid.html

```diff
@@ -484,7 +484,7 @@
           col._currentWidth = 0;
           // Note: _allCells only contains cells which are currently rendered in DOM
           col._allCells.forEach(c => {
-            const cellWidth = Math.ceil(c.getBoundingClientRect().width);
```

Comment:
`element.getBoundingClientRect().width` is affected by the visual scaling of the element whereas `element.offsetWidth` isn't. The original issue was caused by the grid being visually scaled during the dialog opening animation.